internals/overlord/logstate/syslog/syslog.go

```diff
@@ -0,0 +1,300 @@
+package syslog
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/canonical/pebble/internals/servicelog"
+)
+
+const (
+	defaultMaxRequestEntries    = 100
+	defaultSyslogInitialBackoff = 100 * time.Millisecond
```

Comment:
In names like this, "syslog" isn't necessary either, as this is in the `syslog` package:
```suggestion
	initialBackoff = 100 * time.Millisecond
```